docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,183 @@
+.. _complex_numbers-doc:
+
+Complex Numbers
+===============
+
+Complex numbers are numbers that can be expressed in the form :math:`a + bj`, where a and b are real numbers,
+and *j* is a solution of the equation :math:`x^2 = −1`. Complex numbers frequently occur in mathematics and
+engineering, especially in signal processing. Traditionally many libraries (e.g., TorchAudio) have handled
+complex numbers by representing the data in float tensors with shape :math:`(..., 2)` where the last
+dimension contains the real and imaginary values.
+
+Tensors of complex dtypes provide a more natural user experience for working with complex numbers. Operations on
+complex tensors (e.g., :func:`torch.mv`, :func:`torch.matmul`) are likely to be faster and more memory efficient
+than operations on float tensors mimicking them. Operations involving complex numbers in PyTorch are optimized
+to use vectorized assembly instructions and specialized kernels (e.g. LAPACK, cuBlas).
+
+.. note::
+     Spectral operations (e.g., :func:`torch.fft`, :func:`torch.stft` etc.) currently don't use complex tensors but
+     the API will be soon updated to use complex tensors.
+
+.. warning ::
+     Complex Tensors is a beta feature and subject to change.
+
+Creating Complex Tensors
+------------------------
+
+We support two complex dtypes: `torch.cfloat` and `torch.cdouble`
+
+::
+
+     >>> x = torch.randn(2,2, dtype=torch.cfloat)
+     >>> x
+     tensor([[-0.4621-0.0303j, -0.2438-0.5874j],
+          [ 0.7706+0.1421j,  1.2110+0.1918j]])
+
+.. note::
+
+     The default dtype for complex tensors is determined by the default floating point dtype.
+     If the default floating point dtype is `torch.float64` then complex numbers are inferred to
+     have a dtype of `torch.complex128`, otherwise they are assumed to have a dtype of `torch.complex64`.
+
+All factory functions apart from :func:`torch.linspace`, :func:`torch.logspace`, and :func:`torch.arange` are
+supported for complex tensors.
+
+Transition from the old representation
```

Comment:
Can you consolidate all the historical information in this section?

Reply:
Yeah however I think some historical context in the introduction sets a better stage to truly justify `"more natural user experience"` and `"faster and more memory efficient than operations on float tensors mimicking them"`.

Reply:
has anyone made the argument that mimicking complex numbers is a better user experience?  faster?  I haven't personally come across those arguments, my impression is that everyone recognizes the previous behavior as a hack.